Mathlib/Data/List/Defs.lean

```diff
@@ -549,4 +549,31 @@ def replaceIf : List α → List Bool → List α → List α
 #align list.map_with_prefix_suffix List.mapWithPrefixSuffix
 #align list.map_with_complement List.mapWithComplement
 
+/-- `iterate f a n` is `[a, f a, ..., f^[n - 1] a]`. -/
+@[simp]
+def iterate (f : α → α) (a : α) : (n : ℕ) → List α
```

Comment:
Is there precedent for this name in haskell or similar?

Reply:
[iterate](https://hackage.haskell.org/package/base-4.19.1.0/docs/Prelude.html#v:iterate)